airflow/providers/amazon/aws/triggers/glue_crawler.py

```diff
@@ -68,11 +68,8 @@ async def run(self) -> AsyncIterator[TriggerEvent]:
                     break  # we reach this point only if the waiter met a success criteria
                 except WaiterError as error:
                     if "terminal failure" in str(error):
-                        yield TriggerEvent(
-                            {"status": "failure", "message": f"Glue Crawler creation Failed: {error}"}
-                        )
-                        break
+                        raise
                     self.log.info("Status of glue crawl is %s", error.last_response["Crawler"]["State"])
                     await asyncio.sleep(int(self.poll_interval))
 
-        yield TriggerEvent({"status": "success", "message": "Crawl Complete"})
+        yield TriggerEvent.success(self.crawler_name)
```

Comment:
This may be considered a breaking change. It depends on when we will release the next wave of providers we need to be very carful here.

Reply:
ok, I'll split it from this. I see how it's a breaking change, but I believe the glue crawler deferrable has not yet been shipped in a provider package, so if this is merged before the next wave, it should be no trouble.
Also, it was a bad implem (from me) because I didn't return the same thing as the non-deferred version, so we can see this more as a fix.

I'll do 3 PRs:
this fix on glue
the core code change
the conversion to the new method

Reply:
done in #31694

Reply:
The issue is that you can not use
`TriggerEvent.success`
In provider code because the provider must be compatible with Airflow 2.4

Reply:
hmm right...
So you're saying that I can introduce this, but it's gonna be pretty useless because we won't be able to use it until years, when we say that the minimum supported version is 2.7 🤔 
Maybe I can introduce it as an `AwsBaseDeferrableOperator` or something, that way it'd be packaged the same way ?

Reply:
Yes. The way how to use it until 2.7+ (which is not multiple years, but 12 months since we release 2.7) - there has to be a back-compatible way of using it. Unfortunately, vack compatibility of core classses is important.

I think the code to use hte "success" should fall-back to hard-coded version of it.

Reply:
but it's the same for the default method and default param. If airflow is in an old version, `defer` will say it lacks a mandatory parameter, and even if we specify it, we won't find the default method with `getattr` when we want to call it.
None of this can work I think ? 🤔

Reply:
> but it's the same for the default method and default param. If airflow is in an old version, defer will say it lacks a mandatory parameter, and even if we specify it, we won't find the default method with getattr when we want to call it.

welcome to backcompat puzzles @vandonr-amz .  they can be very confusing.  in that particular examplethough, i think you _could_ make `method_name` optional in BaseOperator.defer because, the change itself is backward compatable, and then in providers you just have to use the old signature until the min airflow version is >= 2.x (when the signature was changed)

Reply:
I understand it's the way to go, but it'd be pretty misleading to have an optional param, but you'd have to _know_ that you _have to_ fill it otherwise you break backcompat 🤔